lib/parse.js

```diff
@@ -1146,9 +1156,32 @@ function parse($TEXT, options) {
         --S.in_function;
         S.in_loop = loop;
         S.labels = labels;
+        S.in_generator = current_generator;
         return a;
     }
 
+    function _yield_expression() {
+        if (!is_in_generator()) {
+            throw new JS_Parse_Error("Unexpected yield expression outside generator function",
+                S.prev.file, S.prev.line, S.prev.col, S.prev.pos);
+        }
+        var star = false;
+        var statement = true;
+        var tmp;
+        if (is("operator", "*")) {
+            star = true;
+            next();
+        } else if (can_insert_semicolon() || (
+            is("punc") && ";]),:".indexOf(S.token.value) !== -1
```

Comment:
It's to distinguish if yield must be an expression or if it may be an identifier. Of course, one needs to be able to understand this without looking at this comment so I will look for a fix.